src/submodule.rs

```diff
@@ -33,6 +33,21 @@ impl<'repo> Submodule<'repo> {
         unsafe { crate::opt_bytes(self, raw::git_submodule_branch(self.raw)) }
     }
 
+    /// Perform the clone step for a newly created submodule.
+    ///
+    /// This performs the necessary `git_clone` to setup a newly-created submodule.
+    pub fn clone(&mut self, opts: &mut SubmoduleUpdateOptions<'_>) -> Result<Repository, Error> {
+        unsafe {
+            let mut raw_repo = ptr::null_mut();
+            try_call!(raw::git_submodule_clone(
+                &mut raw_repo,
+                self.raw().as_mut(),
```

Comment:
I think this can be accessed directly.

```suggestion
                self.raw,
```

Reply:
Tested locally and it works.